Tweak.xm

```diff
@@ -19,45 +17,39 @@ static BOOL settingsUpdated = NO;
 }
 %end
 
-%hook SBDashBoardQuickActionsView
-- (void)_layoutQuickActionButtons {
-	%orig;
-	for (UIView *subview in self.subviews) {
-
-		if (subview.frame.origin.x < 50) {
-			CGRect flashlight = subview.frame;
-
-			// Fix for Jumper: The extra buttons are already aligned above the original ones, so don't change their Y position
-			CGFloat flashlightOffset = subview.alpha > 0 ? (flashlight.origin.y - 90 + flashlightY) : flashlight.origin.y;
-
-			flashlight = CGRectMake(46 + flashlightX, flashlightOffset, 50, 50);
-
-			subview.frame = flashlight;
-			[subview sb_removeAllSubviews];
-			[subview init];
-		} else {
-			CGFloat _screenWidth = [UIScreen mainScreen].bounds.size.width;
-			CGRect camera = subview.frame;
-
-			// Fix for Jumper: The extra buttons are already aligned above the original ones, so don't change their Y position
-			CGFloat cameraOffset = subview.alpha > 0 ? (camera.origin.y - 90 + cameraY) : camera.origin.y;
-
-			camera = CGRectMake((_screenWidth - 96) + cameraX, cameraOffset, 50, 50);
-
-			subview.frame = camera;
-			[subview sb_removeAllSubviews];
-			[subview init];
-		}
-	}
+static inline CGFloat GetButtonSize(CGRect screenBounds) {
+	if (screenBounds.size.height >= 812)
+		return 58;
+	if (screenBounds.size.height >= 736)
+		return 50;
+	return 42;
 }
--(void)_addOrRemoveCameraButtonIfNecessary {
-	%orig;
 
-	// Necessary to change the position of the buttons without a respring
-	if (settingsUpdated) {
-		[self _layoutQuickActionButtons];
-		settingsUpdated = NO;
+%hook SBDashBoardQuickActionsView
+- (void)_layoutQuickActionButtons {
+	UIEdgeInsets insets = [self _buttonOutsets];
+	[self.flashlightButton setEdgeInsets:insets];
+	[self.cameraButton setEdgeInsets:insets];
+
+	UIUserInterfaceLayoutDirection layoutDirection = [UIApplication sharedApplication].userInterfaceLayoutDirection;
+	CGRect _referenceBounds = [UIScreen mainScreen]._referenceBounds;
+	CGFloat buttonSize = GetButtonSize(_referenceBounds);
+	CGFloat xOffsetPadding = layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft ? insets.right : insets.left;
+	CGFloat buttonWidth = buttonSize + insets.right + insets.left;
+	CGFloat buttonHeight = buttonSize + insets.top + insets.bottom;
+	CGFloat offsetY = _referenceBounds.size.height - buttonHeight - insets.bottom;
+
+	CGRect flashLightRect;
+	CGRect cameraRect;
+	if (layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
+		flashLightRect = CGRectMake(_referenceBounds.size.width - xOffsetPadding - buttonWidth + flashlightX, offsetY + flashlightY, buttonWidth, buttonHeight);
+		cameraRect = CGRectMake(xOffsetPadding + cameraX, offsetY + cameraY, buttonWidth, buttonHeight);
```

Comment:
Swap the positions of the flashlight and camera for RTL languages as it does natively.